### DISCUSSION TITLE
Feedback: New using directive syntax

### DISCUSSION THREAD

**Discussion by romanowski** [VirtusLab/scala-cli#578] [General]
Hello everyone, the upcoming `0.1.0` release will bring a new syntax for using directives, namely:
- using directives in plain comments (`// using foo bar` and `/* using foo bar */)` is deprecated
- annotation syntax for using directives is deprecated (`@using`) 
- we are introducing special comments for using directives using `>`: `//> using foo bar` and ```/*> using foo bar */```

Old syntaxes will still work in `0.1.x` releases but will results in warnings like this one:

```
[warn] ./a.scala:2:1: This using directive is ignored. Using directive from plain comments are deprecated, please add `>` to each comment.
[warn] // using options "-Xasync"
[warn] ^^^
```
Why do deprecate the comment syntax? With keyword-based directive, in a situation when one would like to comment out such directive (and this is the only directive in the file) simply adding `//` before the directive will just change its syntax when user will expect that the directive is ignored.

There is a bit of time before the `0.1` so we would love to hear some feedback about changes in using directives syntax.

 `0.1.0` [will bring the ability](https://github.com/VirtusLab/scala-cli/pull/548) to run snapshot version directly from scala-cli (or any version if anyone really cares about stability) but until now snapshots can be tested with coursier using snapshot from our master branch. Unfortunately, we publish only jars for nightlies so scala-cli may not be as snappy as in stable releases (it will be JVM app, not a native one as for stable release). To try it out, the following command can be used:

```
cs bootstrap -o ./scala-cli-nightly -r sonatype:snapshots org.virtuslab.scala-cli:cli_2.12:0.0.9+191-gb0cd726d-SNAPSHOT -M scala.cli.ScalaCli
```

It will create `scala-cli-nightly` binary in the current directory that later can be used to try out new syntax.

Please let us know here. what do you think about new syntax.

**Comment by bjornregnell** [comment]
This seems good to me and makes sense. I hadn't though about the "de-commenting interaction", but I think it's worth adding the //> syntax if it is likely that I will be able to use `using foo` without comment that compiles regardless if it is a script or persistently compiled or in metals or in intellij etc. 

One suggestion: Add an example to the deprecation message so users can see how to write it in the new syntax. I mean explicitly show the exact syntax; just saying "add >" is not perhaps open for interpretation and guessing ("huh!? should it be at added at the end of the line or after the //?" etc).

**Reply by romanowski** [reply]
Good point, I've updated [the message](https://github.com/VirtusLab/scala-cli/blob/master/modules/build/src/main/scala/scala/build/preprocessing/ScalaPreprocessor.scala#L342) to be more clear.
